File: LeetCode/Python/#350 Intersection of Two Arrays II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        res = []
        ans = list(set(nums1)&set(nums2))
        for i in ans:
            tmp = min(nums1.count(i),nums2.count(i))
            for j in range(tmp):
                res.append(i)

        return res


class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        nums = {}
        res = []

        for i in nums1:
            nums[i] = nums.get(ele,0)+1
        for i in nums2:
            nums[i] = nums.get(ele,0)-1
            if nums[i]>=0:
                res.append(i)
        return res

File: LeetCode/JavaScript/#350. Intersection of Two Arrays II.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
    if(nums1===nums2) return nums1;
    var len1 = nums1.length;
    var len2 = nums2.length;
    var res = [];
    var pos;
    if(len1<len2){
    	for(let i = 0;i<len1;i++){
    		pos = nums2.indexOf(nums1[i]);
    		if(pos!==-1){
    			res.push(nums1[i]);
    			nums2.splice(pos,1);
    		}
    	}
    }else{
    	for(let i = 0;i<len2;i++){
    		pos = nums1.indexOf(nums2[i]);
    		if(pos!==-1){
    			res.push(nums2[i]);
    			nums1.splice(pos,1);
    		}
    	}
    }
    return res;
};
